src/hooks/screen/useCreateChat.ts

```diff
@@ -35,17 +34,14 @@ const useCreateChat = () => {
       );
       const members = originalMembers || response?.members;
 
+      const promises: Promise<any>[] = [];
+
       members?.forEach(async (member) => {
         const userMember = UserSchema.fromMemberWebsocketObject(member, response?.channel?.id);
         await userMember.saveOrUpdateIfExists(localDb);
-
-        const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
-          response?.channel?.id,
-          uuid(),
-          member
-        );
-        await memberSchema.save(localDb);
       });
+
+      await Promise.all(promises);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a logical issue here. An array of promises (`promises`) is declared but never used. The `forEach` loop is creating asynchronous operations, but these are not being pushed into the `promises` array. Therefore, when `Promise.all(promises)` is executed, it's not waiting for any promises to resolve. 

To fix this, you should push each promise returned by `userMember.saveOrUpdateIfExists(localDb)` into the `promises` array. Here's how you can do that:

```diff
-       members?.forEach(async (member) => {
+       members?.forEach((member) => {
           const userMember = UserSchema.fromMemberWebsocketObject(member, response?.channel?.id);
-          await userMember.saveOrUpdateIfExists(localDb);
+          promises.push(userMember.saveOrUpdateIfExists(localDb));
        });

       await Promise.all(promises);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->